### PR TITLE
[Backport to adb-5x] Fix partition selector under join by keys with different types.

### PR DESCRIPTION
**Pull request by maksm90**
Backport of https://github.com/arenadata/gpdb/pull/167 to adb-5x

### CODE REVIEW

**Comment by maksm90** [comment]
https://github.com/arenadata/gpdb/pull/178 PR to adb-5.x

**Comment by maksm90** [comment]
Test cases are fixed. The patch is ready to be reviewed.

**Review by darthunix — DISMISSED** [review]

File: src/backend/cdb/cdbpartition.c [hunk 1]
```diff
@@ -3856,20 +3854,15 @@ selectListPartition(PartitionNode *partnode, Datum *values, bool *isnull,
...
+                         * descriptor whose number corresponds to current
+                         * partitioning key in partitioned table
 						 */
-						Oid lhstypid = exprTypeOid;
-						if (!OidIsValid(lhstypid))
-						{
-							lhstypid = rhstypid;
-						}
+						Oid rhstypid = c->consttype;
+						Oid lhstypid = tupdesc->attrs[attno - 1]->atttypid;
```
Comment (leskin-in) [inline src/backend/cdb/cdbpartition.c:3865]: The same question as in https://github.com/arenadata/gpdb/pull/167#discussion_r623690992

**Review by leskin-in — DISMISSED** [review]

File: src/test/regress/expected/bfv_partition.out [hunk 2]
```diff
@@ -1938,3 +1938,116 @@ DROP TABLE part_tbl;
 -- start_ignore
 drop schema if exists bfv_partition;
 -- end_ignore
+---
```
Comment (Stolb27) [inline src/test/regress/expected/bfv_partition.out:1937]: I think test addition after cleanup (line 1937) is a bad idea. This test set use `set search_path=bfv_partition;`. As result threre are failed tests with:

```
ERROR:  no schema has been selected to create in
```

I'm going to attach the test output to Jira issue.

**Review by Stolb27 — CHANGES_REQUESTED** [review]
@maksm90 please fix test case

File: src/test/regress/expected/bfv_partition.out (hunk 2 above)
Comment (maksm90) [reply src/test/regress/expected/bfv_partition.out:1937]: Agreed, fixed below.

**Comment by maksm90** [comment]
> @maksm90 please fix test case

Thanks for the notice. Pushed fix, please verify it once again. Unfortunately, I'm not able to do this locally.

**Review by Stolb27 — APPROVED** [review]

**Pull request merged**